File: code/js/src/views/user/TrainerViews/models/setExerciseDetails.ts
class SetExerciseDetails {
    id: number;
    name: string;
    muscleGroup: string[]
    modality: string
    details: Map<string, any>;

    constructor(id: number, name: string, muscleGroup: string[], modality: string,details: Map<string, any>){
        this.id = id;
        this.name = name;
        this.muscleGroup = muscleGroup;
        this.modality = modality;
        this.details = details;
    }
}
export default SetExerciseDetails;

File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/domain/set/model/SetExerciseDetails.kt
package pt.isel.leic.ptgest.domain.set.model

import pt.isel.leic.ptgest.domain.set.ExerciseDetailsType
import pt.isel.leic.ptgest.domain.workout.Modality
import pt.isel.leic.ptgest.domain.workout.MuscleGroup

data class SetExerciseDetails(
    val id: Int,
    val name: String,
    val muscleGroup: List<MuscleGroup>,
    val modality: Modality,
    val details: Map<ExerciseDetailsType, Number>
)
